### ISSUE TITLE
[FEATURE]: gradient checkpoint supports pass an argument dict

### ISSUE THREAD

**Issue by feifeibear** [hpcaitech/ColossalAI#703]
### Describe the feature

In the file colossalai/utils/activation_checkpoint.py
```
def checkpoint(function, activation_offload, *args):
    """Checkpoint the computation while preserve the rng states, modified from Pytorch torch.utils.checkpoint.

    Args:
        function: Describe the forward pass function. It should know how to handle the input tuples.
        args (list): Tuple containing the parameters of the function

    Returns:
        Output of running function with provided args.
    """
    return CheckpointFunction.apply(function, activation_offload, *args)
```

It looks like the checkpoint accepts a tuple as args. If the API supports an extra dict will be more helpful for users.

**Comment by binmakeswell** [comment]
We have updated a lot. This issue was closed due to inactivity. Thanks.
